File: webbluetooth/packages/brilliant-ble/example/clear-display.js
import { BrilliantBle } from 'brilliant-ble';

export async function run() {
  const ble = new BrilliantBle();

  // Connect to Frame
  const deviceId = await ble.connect();

  // Configure print response handler to show Frame output
  const printHandler = (data) => {
    console.log("Frame response:", data);
  };
  ble.setPrintResponseHandler(printHandler);

  // Send a break signal to Frame in case it is in a loop/main.lua
  await ble.sendBreakSignal({showMe: true});

  // Clear the Frame display
  var luaCommand = "frame.display.text('', 1, 1);frame.display.show();print(0)";
  await ble.sendLua(luaCommand, {showMe: true, awaitPrint: true});

  // Wait for a couple of seconds
  await new Promise(resolve => setTimeout(resolve, 2000));

  // Disconnect from Frame
  await ble.disconnect();
};

File: python/packages/brilliant_ble/examples/clear_display.py
import asyncio
import argparse
from brilliant_ble import BrilliantBle, BrilliantDeviceType

async def main():
    parser = argparse.ArgumentParser(description="Connect to a Halo/Frame device over BLE and run this example.")
    parser.add_argument(
        "--name",
        default=None,
        help='exact BLE device name, e.g. "Halo AB" or "Frame 4F"; defaults to the nearest device',
    )
    args = parser.parse_args()
    frame = BrilliantBle()

    try:
        await frame.connect(name=args.name)

        # stop any application, if running, so we can send lua commands
        await frame.send_break_signal()

        # initialize Halo display
        await frame.send_lua("frame.display.power_save(false);print(0)", await_print=True)

        # Clear the display
        if frame._type == BrilliantDeviceType.HALO:
            await frame.send_lua("frame.display.clear();print(0)", await_print=True)
        else:
            await frame.send_lua("frame.display.text('', 1, 1);frame.display.show();print(0)", await_print=True)
        print("Display cleared")

        await frame.disconnect()

    except Exception as e:
        print(f"Not connected to Device: {e}")
        return

if __name__ == "__main__":
    asyncio.run(main())